lib/jsdom/contextify-shim.js

```diff
@@ -1,5 +1,49 @@
+var acorn = require("acorn");
+var findGlobals = require("acorn-globals");
+var escodegen = require("escodegen");
+
 module.exports = function (o) {
   o.getGlobal = function () {
     return o;
   };
+
+  o.run = function (code, filename) {
+    var comments = [], tokens = [];
+    var ast = acorn.parse(code, {
+      ecmaVersion: 6,
+      allowReturnOutsideFunction: true,
+      ranges: true,
+      // collect comments in Esprima's format 
+      onComment: comments,
+      // collect token ranges 
+      onToken: tokens
+    });
+
+    // make sure we keep comments
+    escodegen.attachComments(ast, comments, tokens);
+
+    var globals = findGlobals(ast);
+    for (var i = 0; i < globals.length; ++i) {
+      if (globals[i].name === "window") continue;
```

Comment:
curlies and newline. JSHint should be catching this? Can we JSHint this file now?